mgplvm/kernels/linear.py

```diff
@@ -33,28 +38,52 @@ def __init__(self, n: int, d: int, scale=None, learn_scale=False, Y=None):
 
         if scale is not None:
             _scale = torch.tensor(scale)
-        elif Y is not None:  # <Y^2> = scale * d * <x^2> + <eps^2> = scale * d + sig_noise^2
+        elif (
+                Y is not None
+        ) and learn_scale:  # <Y^2> = scale * d * <x^2> + <eps^2> = scale * d + sig_noise^2
             _scale = torch.tensor(np.sqrt(np.var(
                 Y, axis=(0, 2)) / d)) * 0.5  #assume half signal half noise
         else:
             _scale = torch.ones(n,)  #one per neuron
         self._scale = nn.Parameter(data=_scale, requires_grad=learn_scale)
 
+        _ell = inv_softplus(torch.ones(d))
+        self._ell = nn.Parameter(data=_ell, requires_grad=ard)
+
     def diagK(self, x: Tensor) -> Tensor:
-        diag = (self.scale_sqr[:, None, None] * (x**2)).sum(dim=-2)
+        diag = (self.scale_sqr[:, None, None] *
+                (self.prmtize(x)**2)).sum(dim=-2)
         return diag
 
     def trK(self, x: Tensor) -> Tensor:
-        return self.diagK(x).sum(dim=-1)
+        return self.diagK(self.prmtize(x)).sum(dim=-1)
 
     def K(self, x: Tensor, y: Tensor) -> Tensor:
-        distance = x.transpose(-1, -2).matmul(y)
-        kxy = self.scale_sqr[:, None, None] * distance
+        """
+        Parameters
+        ----------
+        x : Tensor
+            input tensor of dims (... n_samples x n x d x mx)
+        y : Tensor
+            input tensor of dims (... n_samples x n x d x mx)
+
+        Returns
+        -------
+        trK : Tensor
+            trace of kernel K(x,x) with dims (... n)
+        """
+        dot = self.dot(self.prmtize(x), self.prmtize(y))
+        kxy = self.scale_sqr[:, None, None] * dot
         return kxy
 
+    def prmtize(self, x: Tensor) -> Tensor:
```

Comment:
seems a bit weird to call this `prmtize`, maybe `reweight`?

Reply:
also if `ell` is a lengthscale, shouldn't we divide `ell` instead of multiply? or maybe we should change the variable name?

Reply:
reweight sounds good.
I prefer multiplying over dividing since this gives it a nice interpretation of a prior over weights, but agree the name is slightly misleading in this case. I mostly used it for consistency with the stationary kernels. Do you have an alternative suggestion?

Reply:
hmm what about `weights`?

Reply:
could do latent_weights? or latent_prior?
'weights' seems to suggest that it should be something like the FA weights but it's really a prior over weights specified on a latent-to-latent basis?